testsuites/syncgateway/functional/tests/test_conflicts.py

```diff
@@ -271,3 +282,49 @@ def test_winning_conflict_branch_revisions(params_from_base_test_setup, sg_conf_
     assert len(changes_2["results"]) == 1
     assert changes_2["results"][0]["id"] == "test_doc"
     assert changes_2["results"][0]["changes"][0]["rev"] == "7-foo"
+
+
+@pytest.mark.syncgateway
+@pytest.mark.conflicts
+@pytest.mark.parametrize("sg_conf_name, revs_limit", [
+    ('sync_gateway_revs_conflict_configurable', 1),
+    ('sync_gateway_revs_conflict_configurable', 19),
+    ('sync_gateway_revs_conflict_configurable', 'a'),
+    ('sync_gateway_revs_conflict_configurable', -1)
+    # TODO : commenting as revs_limit 0 behavior is going to change, existing behavior start sg successfully , but it will change to sg fails
+    # Enable it once behavior is changed
+    # ('sync_gateway_revs_conflict_configurable', 0)
+])
+def test_invalid_revs_limit_with_allow_conflicts(params_from_base_test_setup, sg_conf_name, revs_limit):
+    """ @summary Verify all borders of revs limit
+    Test case in Excel sheet : https://docs.google.com/spreadsheets/d/1YwI_gCeoBebQKBybkzoAEoXSc0XLReszDA-mPFQapgk/edit#gid=0
+    Covered Test case #2
+    Steps:
+    - Create a doc
+    - Have allow_conflicts to true in sg config
+    - Put revs_limit=1 or any number lower than 20 and restart sync-gateway
+    - Verify it fails
+    - change revs_limit=20 and start sync-gateway
+    - Verify it starts withtout any error
+    """
+
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    mode = params_from_base_test_setup["mode"]
+    no_conflicts_enabled = params_from_base_test_setup["no_conflicts_enabled"]
+
+    if no_conflicts_enabled:
+        pytest.skip('--no-conflicts is enabled, this test needs to create conflicts, so skipping the test')
+
+    sg_conf = sync_gateway_config_path_for_mode(sg_conf_name, mode)
+    clust = cluster.Cluster(cluster_config)
+    clust.reset(sg_conf)
+    temp_cluster_config = copy_to_temp_conf(cluster_config, mode)
+    persist_cluster_config_environment_prop(temp_cluster_config, 'revs_limit', revs_limit, property_name_check=False)
+    status = clust.sync_gateways[0].restart(config=sg_conf, cluster_config=temp_cluster_config)
+    assert status != 0, "Syncgateway started with revs limit 1 when no conflicts disabled"
+
+    # Now change the revs_limit to 20
+    revs_limit = 20
+    persist_cluster_config_environment_prop(temp_cluster_config, 'revs_limit', revs_limit, property_name_check=False)
```

Comment:
Why is property_name_check needed? Why not add it to the valid_props array

Reply:
same comment as above. revs_limit is not mandatory. Based on test requirement, it gets added